src/pynwb/tests/test_hierarchical_behavior_data.py

```diff
@@ -0,0 +1,88 @@
+import shutil
+from datetime import datetime
+from pathlib import Path
+from tempfile import mkdtemp
+
+import numpy as np
+from dateutil import tz
+from hdmf.testing import TestCase
+from pandas.testing import assert_frame_equal
+from pynwb import NWBFile, NWBHDF5IO
+from pynwb.epoch import TimeIntervals
+
+from ndx_hierarchical_behavioral_data import HierarchicalBehavioralTable
+
+
+class TestHierarchicalBehavioralTable(TestCase):
+    def setUp(self):
+        self.test_dir = Path(mkdtemp())
+
+        self.nwbfile = NWBFile(
+            session_description="session_description",
+            identifier="identifier",
+            session_start_time=datetime.now().astimezone(tz=tz.gettz("US/Pacific")),
+        )
+        self.nwbfile_path = self.test_dir / "test.nwb"
+
+        self.lower_tier_table = TimeIntervals(
+            name="Words",
+            description="The intervals for the lowest hierarchy.",
+        )
+
+        self.lower_tier_table.add_column(
+            name="label", description="The label for this table."
+        )
+
+        self.lower_tier_table.add_row(start_time=0.3, stop_time=0.5, label="The")
+        self.lower_tier_table.add_row(start_time=0.7, stop_time=0.9, label="First")
+        self.lower_tier_table.add_row(start_time=1.3, stop_time=3.0, label="Sentence")
+        self.lower_tier_table.add_row(start_time=4.0, stop_time=5.0, label="And")
+        self.lower_tier_table.add_row(start_time=6.0, stop_time=7.0, label="Another")
+
+    def tearDown(self):
+        shutil.rmtree(self.test_dir)
+
+    def test_roundtrip(self):
+        sentences_table = HierarchicalBehavioralTable(
+            name="Sentences",
+            description="The behavioral table.",
+            lower_tier_table=self.lower_tier_table,
+        )
+        sentences_table.add_interval(
+            label="Sentence1",
+            next_tier=[0, 1, 2],
+        )
+        sentences_table.add_interval(
+            label="Sentence2",
+            next_tier=[3, 4],
+        )
+
+        sentences_dataframe = sentences_table.to_hierarchical_dataframe()
```

Comment:
I like that you included this in the tests but I think it would be better in a separate test. Generally each method should have it's own test. That way we can see what methods are broken by what tests failed.